### PR TITLE
feat: [BB-239] implement formik for crops form

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Implemented Formik for Crops Form, doesn't completely follows the designer's mockups but the forms are aligned mobile and desktop. 

> [!NOTE]  
> You can check the console that the data has been pushed to the farm information object

Mockup Reference:
Group 105
Group 60(1)

> [!WARNING]  
> If you add two fields, once you click add on the second field with the corresponding data, it would be added on the first field.
> Please do note, that this isn't a bug but will be fixed on the next ticket that I'll be working

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-261-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-261-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by kcaparas1630** [comment]
Lines might be long, but it's only due to a file move.

File: frontend/src/Commons/Forms/InputModules/Crops/CropsButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import Button from '@Commons/Button/Button';
+import {
+  StyledButtonGroupContainer,
+  StyledButtonContainer,
+  StyledAddCancelButtonContainer,
+} from '../../../Button/FieldButtonGroup.styles';
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsButtonGroup.tsx:7]: Maybe use `@Commons/Button/...`

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+import {
+  StyledCropsSmallGroup,
+  StyledCropsLargeGroup,
+  StyledAddCancelButtonGroup,
+} from './CropsInfo.styles';
+import {
+  StyledFarmInfo,
+  StyledButtonGroupContainer,
+  StyledAreaContainer,
+} from '../../../FormStyles.styles';
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:41]: Maybe use `@Commons/...`
Is there a better directory for `FormStyles.styles` though?

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    }
+    if (values.distanceBtwnPlants === '0.9' && values.distanceBtwnRows === '2.7') {
+      resultString += '(3ft x 9ft)';
+    }
+    if (values.distanceBtwnPlants === '0.9' && values.distanceBtwnRows === '3.0') {
+      resultString += '(3ft x 10ft)';
+    }
+
+    return resultString;
+  };
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:99]: Great candidate for a switch if you want to optimize this a bit.

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        cropId: values.cropId,
+        yield: values.yield,
+        plantAgeYears: values.plantAgeYears,
+        numberOfPlantsPerAcre: values.numberOfPlantsPerAcre,
+        distanceBtwnPlantsRows: `${values.distanceBtwnPlants} m x ${values.distanceBtwnRows} m ${distanceBtwnPlantsRowsDataAppend(values)}`,
+        willPlantsBePruned: values.willPlantsBePruned,
+        whereWillPruningsGo: values.whereWillPruningsGo,
+        willSawdustBeApplied: values.willSawdustBeApplied,
+      });
+      console.log(farmInfo);
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:114]: :eyes:

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx [hunk 5]
```diff
@@ -38,7 +38,7 @@ import {
   StyledSelectContainer,
   InputFieldsGroup,
   SingleInputField,
-} from './FieldsAndSoil.style';
+} from '../../../FormStyles.styles';
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx:41]: :eyes:

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx [hunk 6]
```diff
@@ -178,7 +178,7 @@ const FieldsAndSoilComponent: FC<InputModuleProps> = ({
                     name="FieldName"
                     type="text"
                   />
-                  <StyledAreaContainer>
+                  <StyledAreaContainer formCrops={false}>
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx:181]: What's up with this? Could you make `formCrops` optional so we don't need to pass this prop everywhere? Or is this necessary here?

**Review by GDamaso — COMMENTED** [review]
Very minor thingies. Thank you for your effort in this module, it was a hard one to debug. Glad you stuck to it and solved it so gracefully.
Good job man! Everyday improving a little bit :rocket:

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx (hunk 3 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:99]: Makes sense alright I'll do switch

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx (hunk 4 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:114]: Forgot lol! Good catch

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx (hunk 6 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx:181]: It's necesarry. Unless there's a way to make it default to false on emotion.

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx (hunk 2 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:41]: Commons I guess

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx (hunk 4 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:114]: But then I forgot it was necessary for this ticket lmao

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx [hunk 7]
```diff
@@ -42,7 +42,7 @@ const FieldsButtonComponent: FC<ButtonComponentProps> = ({
         </Button>
       </StyledNewFieldButtonController>
     </StyledNewFieldButtonContainer>
-    <StyledAddCancelButtonContainer>
+    <StyledAddCancelButtonContainer formCrops={false}>
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx:45]: :eyes:

**Review by GDamaso — APPROVED** [review]
Good work :rocket:

**Pull request merged**